### PR TITLE
#37 WaitForWallet class doesn't wait for job status in a proper way

### PR DESCRIPTION
**Pull request by OleksandrSankin**
Issue: https://github.com/amihaiemil/zold-java-client/issues/37

WaitForWallet class waits for HTTP status 200 that is not correspond to Zold protocol and leads to the error that described in the issue above. It should wait for HTTP body "OK" according to Zold protocol.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#38](https://github.com/amihaiemil/zold-java-client/issues/38) is now in scope, role is `REV`

**Comment by OleksandrSankin** [comment]
@g4s8 Description added

File: src/main/java/com/amihaiemil/zold/WaitForWallet.java [hunk 1]
```diff
@@ -87,25 +93,92 @@ public Wallet handleResponse(final HttpResponse httpResponse)
...
+                        throw new IllegalStateException(
+                                "Can't wait for job " + jobId + " during "
+                                + this.timeToWait.toMinutes() + " minutes");
+                    }
+
+                } while (true);
+
                 return new RtWallet(this.client, this.baseUri);
             } catch (final InterruptedException | URISyntaxException ex) {
                 throw new IllegalStateException(
```
Comment (g4s8) [inline src/main/java/com/amihaiemil/zold/WaitForWallet.java]: @OleksandrSankin `InterruptedException` is not handled correctly here, catch block should call `Thread.currentThread().interrupt()` if method can handle this kind of exception or rethrow it as-is if can't, see https://stackoverflow.com/q/3976344/1723695

I don't think that throwing `IllegalStateException` when thread was interrupted is a right design.
// cc @amihaiemil

File: src/main/java/com/amihaiemil/zold/WaitForWallet.java [hunk 2]
```diff
@@ -59,6 +60,11 @@
      */
     private final URI baseUri;
 
+    /**
+     * Time to wait for job to be finished.
+     */
+    private final Duration timeToWait = Duration.ofMinutes(5);
```
[the remark this answers is not in the document]
Comment (OleksandrSankin) [reply src/main/java/com/amihaiemil/zold/WaitForWallet.java]: Fixed - make static

File: src/main/java/com/amihaiemil/zold/WaitForWallet.java [hunk 3]
```diff
@@ -87,25 +93,92 @@ public Wallet handleResponse(final HttpResponse httpResponse)
...
-                ).addParameter("id", jobId);
-                final HttpGet status = new HttpGet(uri.build());
-                this.client.execute(
-                    status,
-                    new MatchStatus(
-                        status.getURI(),
-                        HttpStatus.SC_OK
-                    )
-                );
+                final Duration start = Duration.ofNanos(System.nanoTime());
```
[the remark this answers is not in the document]
Comment (OleksandrSankin) [reply src/main/java/com/amihaiemil/zold/WaitForWallet.java]: Fixed - use Instant object

File: src/main/java/com/amihaiemil/zold/WaitForWallet.java (hunk 1 above)
Comment (amihaiemil) [reply src/main/java/com/amihaiemil/zold/WaitForWallet.java]: @g4s8 we'll fix that in #39, thanks :)

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by amihaiemil** [comment]
@rultor merge it again...

**Comment by 0crat** [comment]
The job [#38](https://github.com/amihaiemil/zold-java-client/issues/38) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**